IronFoundry.Container.Shared/Utilities/FileSystemEffectiveAccessComputer.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Security.AccessControl;
+using System.Security.Principal;
+
+namespace IronFoundry.Container.Utilities
+{
+    public class FileSystemEffectiveAccessComputer
+    {
+        public FileSystemRights ComputeAccess(string path, IdentityReference identity)
+        {
+            var reader = new FileSystemSecurityDescriptorReader(path);
+            var descriptor = reader.GetSecurityDescriptor();
+            var effectiveAccess = new EffectiveAccessComputer();
+            var access = effectiveAccess.ComputeAccess(descriptor, identity);
+            var rights = (FileSystemRights) access;
```

Comment:
You can use the EffectiveAccessComputer code to check for access to resources other than the filesystem (e.g. registry keys). 
An ACCESS_MASK's first 16 bits are specific to the resource you requested the check for.  The higher 16 bits are generic.